addons/mail_client_extension/controllers/main.py

```diff
@@ -0,0 +1,323 @@
+        return auth_code
+
+    def _iap_enrich(self, domain):
+        enriched_data = {}
+        try:
+            response = request.env['iap.enrich.api']\
+                .with_user(request.env.user)\
+                ._request_enrich({domain: domain}) # The key doesn't matter
+        except odoo.addons.iap.models.iap.InsufficientCreditError as ice:
+            _logger.info('Enrich request: failed because of credit')
+            enriched_data['enrichment_info'] = {'type': 'insufficient_credit', 'info': request.env['iap.account'].with_user(request.env.user).get_credits_url('reveal')}
+        except Exception as e:
+            enriched_data["enrichment_info"] = {'type': 'other', 'info': 'Unknown reason'}
+        else:
+            enriched_data = response.get(domain)
+            _logger.info('Enrich request: success')
+            if not enriched_data:
+                enriched_data = {'enrichment_info': {'type': 'no_data', 'info': 'The enrichment API found no data for the email provided.'}}
+        return enriched_data
+
+    def _iap_enrich_partner_from_response(self, env, iap_data, partner):
+        enriched = {}
+        if not partner.name and iap_data.get("name"):
+            enriched["name"] = iap_data.get("name")
+        if not partner.street and iap_data.get("location"):
+            enriched["street"] = iap_data.get("location")
+        if not partner.city and iap_data.get("city"):
+            enriched["city"] = iap_data.get("city")
+        if not partner.zip and iap_data.get("postal_code"):
+            enriched["zip"] = iap_data.get("postal_code")
+        if not partner.phone and iap_data.get('phone_numbers'):
+            enriched['phone'] = iap_data['phone_numbers'][0]
+        if not partner.website and iap_data.get("domain"):
+            enriched['website'] = iap_data['domain']
+        country = partner.country_id
+        if not partner.country_id and iap_data.get('country_code'):
+            country = env['res.country'].search([('code', '=', iap_data['country_code'].upper())])
+            enriched['country_id'] = country.id
+        if not partner.state_id and country and iap_data.get('state_code'):
+            state = env['res.country.state'].search([
+                ('code', '=', iap_data['state_code']),
+                ('country_id', '=', country.id)
+            ])
+            enriched['state_id'] = state.id
+
+        partner.iap_enrich_info = iap_data
+        # TODO: website
+        partner.write(enriched)
+
+    @http.route('/mail_client_extension/modules/get', type="json", auth="outlook", csrf=False, cors="*")
+    def modules_get(self,  **kwargs):
+        modules = request.env['ir.module.module'].sudo().search([('name', 'in', ['contacts', 'crm']), ('state', '=', 'installed')]).mapped('name')
+        response = {
+            'authorized': True,
+            'modules': modules
+        }
+
+        return response
```

Comment:
```suggestion
        return ['contacts', 'crm']
```
as noted privately, module depends on crm which depends on contacts, so at this point we know both are available.

Coudl also remove method entirely but this way flow on the client side doesn't need to change, and even if we add a few extensions this won't need to change until and unless the dependency on crm is removed.